feature/gribi/otg_tests/mpls_in_udp/README.md

```diff
@@ -1,153 +1,185 @@
-# TE-18.1: gRIBI MPLS in UDP Encapsulation
+# TE-18.1: gRIBI MPLS-in-UDP Encapsulation
 
-Create AFT entries using gRIBI to match on next hop group in a
-network-instance and encapsulate the matching packets in MPLS in UDP with outer header as IPv6 Header.
+## Summary
 
-The MPLS in UDP encapsulation is expected to follow
-[rfc7510](https://datatracker.ietf.org/doc/html/rfc7510#section-3),
-but relaxing the requirement for a well-known destination UDP port.  gRIBI is
-expected to be able to set the destination UDP port.
+Test MPLS-in-UDP encapsulation using gRIBI to create AFT entries that
+match IPv6 traffic and encapsulate it with MPLS labels inside UDP
+packets with IPv6 outer headers. The test validates that the DUT
+correctly implements RFC 7510 MPLS-in-UDP encapsulation with
+configurable UDP destination ports.
 
 ## Topology
 
-* [`featureprofiles/topologies/atedut_2.testbed`](https://github.com/openconfig/featureprofiles/blob/main/topologies/atedut_2.testbed)
+ATE port-1 \<——\> port-1 DUT
 
-## Test setup
+DUT port-2 \<——\> port-2 ATE
 
-TODO: Complete test environment setup steps
+- [`featureprofiles/topologies/atedut_2.testbed`](https://github.com/openconfig/featureprofiles/blob/main/topologies/atedut_2.testbed)
 
-inner_ipv6_dst_A = "2001:aa:bb::1/128"
-inner_ipv6_dst_B = "2001:aa:bb::2/128"
-inner_ipv6_default = "::/0"
+## Test Configuration
 
-ipv4_inner_dst_A = "10.5.1.1/32"
-ipv4_inner_dst_B = "10.5.1.2/32"
-ipv4_inner_default = "0.0.0.0/0"
+The test uses the following key parameters:
 
-outer_ipv6_src =      "2001:f:a:1::0"
-outer_ipv6_dst_A =    "2001:f:c:e::1"
-outer_ipv6_dst_B =    "2001:f:c:e::2"
-outer_ipv6_dst_def =  "2001:1:1:1::0"
-outer_dst_udp_port =  "6635"
-outer_dscp =          "26"
-outer_ip-ttl =        "64"
+    # MPLS-in-UDP configuration
+    mpls_label = 100
+    outer_ipv6_src = "2001:db8::1"
+    outer_ipv6_dst = "2001:db8::100"
+    outer_dst_udp_port = 6635  # RFC 7510 standard MPLS-in-UDP port
+    outer_ip_ttl = 64
+    outer_dscp = 10
+    inner_ipv6_prefix = "2001:db8:1::/64"
 
-## Procedure
-
-### TE-18.1.1 Match and Encapsulate using gRIBI aft modify
+    # Static ARP configuration for gRIBI next hop resolution
+    magic_ip = "192.168.1.1"
+    magic_mac = "02:00:00:00:00:01"
 
-#### gRIBI RPC content
+## Baseline Setup
 
-The gRIBI client should send this proto message to the DUT to create AFT
-entries.
+- Apply VRF selection policy to DUT port-1 to route traffic to the
+  DEFAULT network instance
+- Configure static ARP entries for gRIBI next hop resolution
+  (device-specific)
+- Set up basic routing infrastructure to forward encapsulated packets to
+  port-2
 
-```proto
-#
-# aft entries used for network instance "NI_A"
-IPv6Entry {2001:DB8:2::2/128 (NI_A)} -> NHG#100 (DEFAULT VRF)
-IPv4Entry {203.0.113.2/32 (NI_A)} -> NHG#100 (DEFAULT VRF) -> {
-  {NH#101, DEFAULT VRF}
-}
+Using gRIBI, install the following AFT entries:
```

Comment:
Since this action is part of the primary subject test criteria, it should be present as a subtest, not in the "test environment setup".     ie:  `## TE-18.1.1 - Using gRIBI, install AFT entries`